internal/iopool/pool.go

```diff
@@ -0,0 +1,25 @@
+package iopool
+
+import "sync"
+
+var _maxCapacity = 1024 * 50 // The max capacity for a buffer is 32 KiB
+var _defaultCapacity = 1024 * 32
+var _pool = sync.Pool{
+	New: func() interface{} {
+		return newBuffer(_defaultCapacity)
+	},
+}
+
+// Get returns a new Byte Buffer from the buffer pool
+// that has been reset
+func get() *buffer {
+	buf := _pool.Get().(*buffer)
+	return buf
+}
+
+// Put returns byte buffer to the buffer pool
+func put(buf *buffer) {
```

Comment:
do you expect to use this anywhere other than `Copy`? If so, we can keep it, but otherwise, I think you can simplify this whole change to:

```go
type buffer struct {
  b []byte
}

var _copyPool = sync.Pool{
  New: func() interface{} {
    return &buffer{make([]byte, _copyBufSize)}
  },
}

func Copy(dst io.Writer, src io.Reader) (int64, error) {
  buf := _copyPool.Get().(*buffer)
  written, err := io.CopyBuffer(dst, src, buf.b)
  _copyPool.Put(buf)
  return written, err
}
```

You don't need to check a max capacity since `CopyBuffer` will never change the buffer.